src/main/java/io/cryostat/rules/Rule.java

```diff
@@ -60,10 +65,14 @@
     private final int maxSizeBytes;
 
     Rule(Builder builder) throws MatchExpressionValidationException {
-        this.name = sanitizeRuleName(requireNonBlank(builder.name, Attribute.NAME));
+        this.eventSpecifier = builder.eventSpecifier;
+        if (isArchiver()) {
+            this.name = builder.name;
```

Comment:
Why aren't archive names sanitized?
```
{
    "name": "unsanitized name",    
    "eventSpecifier": "archive",
    "matchExpression": "target.labels.app=='MyApp'"
}
```

Reply:
They're actually just optional on archive rules, since they aren't used to create any recording name and aren't persisted. You can omit the `name` field entirely when creating an archive rule.